bgpd/bgp_vty.c

```diff
@@ -11074,6 +12018,84 @@ static void bgp_show_peer(struct vty *vty, struct peer *p, bool use_json,
 	}
 }
 
+static int bgp_show_neighbor_graceful_restart(struct vty *vty,
+					struct bgp *bgp,
+					enum show_type type,
+					union sockunion *su,
+					const char *conf_if, afi_t afi,
+					bool use_json, json_object *json)
+{
+	struct listnode *node, *nnode;
+	struct peer *peer;
+	int find = 0;
+	safi_t safi = SAFI_UNICAST;
+	json_object *json_neighbor = NULL;
+
+	for (ALL_LIST_ELEMENTS(bgp->peer, node, nnode, peer)) {
+
+		if (!CHECK_FLAG(peer->flags, PEER_FLAG_CONFIG_NODE))
+			continue;
+
+		if ((peer->afc[afi][safi]) == 0)
+			continue;
+
+		if (use_json)
+			json_neighbor = json_object_new_object();
+
+		if (show_all == type) {
```

Comment:
I understand that this is a safeguard against `type = show_all` cases but can we have `type == show_all` here and elsewhere?

Reply:
ok done